config/config.go

```diff
@@ -584,18 +598,57 @@ type MarathonSDConfig struct {
 	XXX map[string]interface{} `yaml:",inline"`
 }
 
+// KubernetesSDConfig is the configuration for Kubernetes service discovery.
+type KubernetesSDConfig struct {
+	Server           string   `yaml:"server"`
+	NodeMetricsPort  int      `yaml:"node_metrics_port,omitempty"`
+	InCluster        bool     `yaml:"in_cluster,omitempty"`
+	BearerTokenFile  string   `yaml:"bearer_token_file,omitempty"`
+	Username         string   `yaml:"username,omitempty"`
+	Password         string   `yaml:"password,omitempty"`
+	Insecure         bool     `yaml:"insecure,omitempty"`
+	CertFile         string   `yaml:"cert_file,omitempty"`
+	KeyFile          string   `yaml:"key_file,omitempty"`
+	CAFile           string   `yaml:"ca_file,omitempty"`
+	RetryInterval    Duration `yaml:"retry_interval,omitempty"`
+	RequestTimeout   Duration `yaml:"request_timeout,omitempty"`
+	DiscoverServices bool     `yaml:"discover_services,omitempty"`
```

Comment:
We generally try to avoid adding functionality that's already possible with relabelling. This helps keep the number of configuration options and permutations sane.